saleor/graphql/order/mutations/draft_order_delete.py

```diff
@@ -54,12 +55,27 @@ def clean_instance(cls, info: ResolveInfo, instance):
     def perform_mutation(cls, _root, info: ResolveInfo, /, **data):
         order = cls.get_instance(info, **data)
         manager = get_plugin_manager_promise(info.context).get()
-        with traced_atomic_transaction():
-            response = super().perform_mutation(_root, info, **data)
-            call_order_event(
-                manager,
-                WebhookEventAsyncType.DRAFT_ORDER_DELETED,
-                order,
+        try:
+            with traced_atomic_transaction():
+                response = super().perform_mutation(_root, info, **data)
+                call_order_event(
+                    manager,
+                    WebhookEventAsyncType.DRAFT_ORDER_DELETED,
+                    order,
+                )
+        except ProtectedError as e:
```

Comment:
What types of models can be protected when deleting draft orders? IMO, if users want to remove a draft order, they shouldn't care about related objects; it's on us to delete them accordingly. Or is it a business requirement to inform them about related objects, and they should handle deleting them manually?

Reply:
The protected objects are: `Payment` and `TransactionItem`. Just for now to stop the 500 errors we are introducing the error handling and separately we want to expend logic with e.g. flags to control the flow. The future is yet to be agreed, but we need to ensure the control over it, the same way as we do in checkout, that why for now it's just proper error handling for the users.

Reply:
Right now, the error message reveals the internals of Saleor (names of models) which is irrelevant to the user. If you want to keep it that way I would at least change the error message to some generic message explaining why the order cannot be removed.

Reply:
I'm going to move the error-handling to `clean_input` - this way it will more readable code and easier to properly provide error message.